components/mpas-ocean/src/shared/mpas_ocn_tracer_hmix_redi.F

```diff
@@ -432,21 +499,48 @@ subroutine ocn_tracer_hmix_Redi_tend(meshPool, layerThickness, layerThickEdgeMea
          end do
 
          if (maxLevelCell(iCell) .ge. config_Redi_min_layers_diag_terms) then
+            ! term 3: d/dz div( S grad psi )
             do k = minLevelCell(iCell), maxLevelCell(iCell)
+               kUp = max(minLevelCell(iCell), k - 1)
+               kDn = min(maxLevelCell(iCell), k + 1)
                do iTr = 1, ntracers
-                  tend(iTr, k, iCell) = tend(iTr, k, iCell) + rediKappaCell(iCell)*2.0_RKIND* &
-                                        (RediKappaScaling(k, iCell)* &
-                                         redi_term3_topOfCell(iTr, k, iCell) - &
-                                         RediKappaScaling(k + 1, iCell)*redi_term3_topOfCell(iTr, k + 1, iCell))
+                  ! harmonic-mean; not strictly monotone, but smooth
+                  scalUp = 2.0_RKIND * &
+                     redi_flux_scale(iTr, kUp, iCell) * &
+                     redi_flux_scale(iTr, k, iCell) / ( &
+                     redi_flux_scale(iTr, kUp, iCell) + &
+                     redi_flux_scale(iTr, k, iCell) + 1.0E-16_RKIND)
+
+                  scalDn = 2.0_RKIND * &
+                     redi_flux_scale(iTr, kDn, iCell) * &
+                     redi_flux_scale(iTr, k, iCell) / ( &
+                     redi_flux_scale(iTr, kDn, iCell) + &
+                     redi_flux_scale(iTr, k, iCell) + 1.0E-16_RKIND)
+
+                  tend(iTr, k, iCell) = tend(iTr, k, iCell) + &
+                     rediKappaCell(iCell) * ( &
```

Comment:
Note that the factor of two in term 3 was also removed here. I just wanted to mark that in this PR, as it would be easy to miss. The factor of two in lines 508 and 514 above are for the harmonic mean only, and is unrelated.